src/components/PrivateRoute.jsx

```diff
@@ -0,0 +1,23 @@
+import { Redirect, Route } from 'react-router-dom';
+import { useAuth } from '../hooks/useAuth';
+
+export function PrivateRoute({ children, ...rest }) {
+  let auth = useAuth();
+  return (
+    <Route
+      {...rest}
+      render={({ location }) =>
+        auth.currentUser.email ? (
+          children
+        ) : (
+          <Redirect
+            to={{
+              pathname: '/auth',
+              state: { from: location },
+            }}
```

Comment:
Excellent work using the location property to send the user to the last location they visited prior to being redirected.